custom_containers/pytorch/mnist/trainer/mnist.py

```diff
@@ -0,0 +1,195 @@
+      hyperparameter_metric_tag='my_loss',
+      metric_value=test_loss,
+      global_step=epoch)
+
+
+def main():
+  # Training settings
+  parser = argparse.ArgumentParser(description='PyTorch MNIST Example')
+  parser.add_argument(
+      '--batch-size',
+      type=int,
+      default=64,
+      metavar='N',
+      help='input batch size for training (default: 64)')
+  parser.add_argument(
+      '--test-batch-size',
+      type=int,
+      default=1000,
+      metavar='N',
+      help='input batch size for testing (default: 1000)')
+  parser.add_argument(
+      '--epochs',
+      type=int,
+      default=10,
+      metavar='N',
+      help='number of epochs to train (default: 10)')
+  parser.add_argument(
+      '--lr',
+      type=float,
+      default=0.01,
+      metavar='LR',
+      help='learning rate (default: 0.01)')
+  parser.add_argument(
+      '--momentum',
+      type=float,
+      default=0.5,
+      metavar='M',
+      help='SGD momentum (default: 0.5)')
+  parser.add_argument(
+      '--no-cuda',
+      action='store_true',
+      default=False,
+      help='disables CUDA training')
+  parser.add_argument(
+      '--seed',
+      type=int,
+      default=1,
+      metavar='S',
+      help='random seed (default: 1)')
+  parser.add_argument(
+      '--log-interval',
+      type=int,
+      default=10,
+      metavar='N',
+      help='how many batches to wait before logging training status')
+  parser.add_argument(
+      '--model-dir',
+      default=None,
+      help='The directory to store the model')
+
+  args = parser.parse_args()
+  use_cuda = not args.no_cuda and torch.cuda.is_available()
+
+  torch.manual_seed(args.seed)
+
+  device = torch.device('cuda' if use_cuda else 'cpu')
+
+  kwargs = {'num_workers': 1, 'pin_memory': True} if use_cuda else {}
+  train_loader = torch.utils.data.DataLoader(
+      datasets.MNIST(
+          '../data',
+          train=True,
+          download=True,
+          transform=transforms.Compose([
+              transforms.ToTensor(),
+              transforms.Normalize((0.1307,), (0.3081,))
+          ])),
+      batch_size=args.batch_size,
+      shuffle=True,
+      **kwargs)
+  test_loader = torch.utils.data.DataLoader(
+      datasets.MNIST(
+          '../data',
+          train=False,
+          transform=transforms.Compose([
+              transforms.ToTensor(),
+              transforms.Normalize((0.1307,), (0.3081,))
+          ])),
+      batch_size=args.test_batch_size,
+      shuffle=True,
+      **kwargs)
+
+  model = Net().to(device)
+  optimizer = optim.SGD(model.parameters(), lr=args.lr, momentum=args.momentum)
+
+  for epoch in range(1, args.epochs + 1):
+    train(args, model, device, train_loader, optimizer, epoch)
+    test(args, model, device, test_loader, epoch)
+
+  if args.model_dir:
+    tmp_model_file = os.path.join('/tmp', MODEL_FILE_NAME)
+    torch.save(model.state_dict(), tmp_model_file)
+    subprocess.check_call([
```

Comment:
Do we need any special configuration for authentication when we are using gsutil cp ?

Reply:
I think that because this is run on CMLE (or that's the purpose of the sample) it should be fine.
I've been using google-cloud-storage python library in mine and haven't had to do anything.

Reply:
that makes sense, Thanks.

Reply:
When running on CMLE, `gsutil` will use the default credential populated by GCE, which is the CMLE service account credential.